### ISSUE TITLE
nations

### ISSUE THREAD

**Issue by boy0001** [IntellectualSites/PlotSquared#31]
This is kind of a big idea (or several). The main concept is to have a system of alliance,war etc with people having their plots join a nation or group of other player's plots.

This is mostly a survival oriented idea, and as such a natural world generator would be required. 
- Possibly claiming a plot will place roads around it (so it can look like a civilized nation)
- Plot merging would still be a thing, and it would be per player.

So what would you get from having your plot being part of a nation?
- Protection, so your plot is less likely to be taken over by enemy nations or barbarians (need to work out war system)
- You won't get attacked by other players in your nation, or it's allies.
- get some bonuses from nation powers (still need to figure this thing out as well)
- Nations can set default flags which all member plots must have
- Nations can set what plot group all nation members would be added as to other plots
  (e.g. nation members are automatically trusted for all plots in the nation)
- Nations can have a tax per plot, or just a single up front fee to join your plot/s with the nation

Economy system:
- Cost for nation to build defenses and expand land
- Nations have a creation cost and an upkeep cost
- This upkeep must be paid or the nation falls apart (usually they will want a bank)
- People can simply donate to the nation, or the nation can set a set tax per member or per plot
- Nations could also have an upfront joining fee and survive on enough people continually joining/leaving the nation

Nation hierarchy:
- President (total control)
- Elected leaders (if the majority members are dissatisfied with the leader they can attempt to overthrow the leader?)
- Councilor (can add/remove people from the nation, and can also build everywhere, can do stuff with the nation's money like withdraw, deposit, spend. Admins can also change the nations diplomacy shit. CANNOT appoint titles)
- Member (has a plot in the nation, pays taxes to nation bank)
- Peasant (is a helper/trusted in one of the plots in the nation, may pay tax to landlord. Has no rights within the actual nation)

Nation powers:
- We could have a bunch "technologies" which can be "researched" in a nation. Researching something will unlock new tools for attack/defence/building/trading within and with other nations.
- We could also have a skill system in which all nation members get maybe certain attack bonuses, or mining bonuses etc based on the nations chosen skill

War system:
- Mostly unsure about this

We would want to have something fair which makes it difficult to actually take over someone's plot, but also not impossible even if they are offline.

BARBARIANS: 
- We don't want to force people into joining a nation, so people could be barbarians which attack the gigantic nations. 
- Can still build without being in a nation
- Barbarians wouldn't have any defend-able land as they would most likely be hard to defend against large nations
- Steal money from opponents treasury by killing players if not part of a nation

We also don't want an entire nation to suddenly fall overnight while some players have been offline. That wouldn't be any fun.
- Force invasions to only take place on the outer plots (so anything exposed to the wilderness can be attacked)

Economy based war system:
- Nations could spend money in the bank to generate defenses like walls (more money = stronger walls)
- Other nations could spend money on (imaginary) siege to try to destroy this wall
- Players could brute force the defences by mining it (each block you destroy takes down the health of the wall)
- Money can be spent every day to rebuild the wall (every real day) however it should be costly to do a partial repair.
- Alternatively blocks could be used to build/repair a wall (through some command interface)
- The plot should also have it's own default health (which should be much smaller than a full wall, but still  decent enough to stop something)
- Nations must pay a price to finally claim a plot once they breach the defences

PVP system:
- Killing players lowers a nations power (like factions)
- If power has decreased by a certain number of units any enemy can take advantage of this and instantly destroy all the defenses for a plot.
- This system would be useful for barbarians who want to raid a single plot, however PVP alone shouldn't allow an inferior band of barbarians (who have no power to loose) from taking over an entire nation.

Self balancing war system:
We don't want a runaway effect in which a nation that is ahead, continuously becomes more and more powerful than it's enemies. Some mechanics need to be in place to make small nation competitive against stronger ones.
- Technology stealing if opponent has researched something
- Money is generated from alliances (from automatic trade). The amount received will depend on the size of the other nation. I.e. the smaller nation will receive more income from automatic trading than the bigger one.

Intra-nation contracts:
- A nation may create contracts e.g. 10 stacks of cobble is needed for a new section of wall. People who help complete this contract in exchange for money.

Money generation:
- Selling to server economy system (stock essentials shop would work)
- Certain buildings will generate money
- Selling to other players
- Users are given the money generated from alliances
- killing players / capturing plots

========== Would you be able to help me make this at some point Citymonstret?

TL;DR - sorta like towny mixed with factions but scales better, and will have a better designed war system.

**Comment by oDeathwingo** [comment]
Also nationwide/custom plot borders so people know its that nation
Maybe custom roads for that nation ?
and custom fee to join/leave the nation and maybe a rank based plot permissions (nation leaders can configure?) 
perms for (can create nation,can join x nations,can be ally, can rankup to xx,...)
nation special blocks ?(only that nation use them,for more themed nation action)
player level to raid or plots have power levels? (so a peasent can't raid a leader plot)
plot capturing (so the other peasents who do not join to a nation can/may get raided and if too poor captured/taxed till the raider nation gets weaker?)
Nation wise power levels and boosts (so invading nation will have less power in other nation's plots)
Starting protection (for everyone so that they do not get crushed by nations right away)

**Comment by oDeathwingo** [comment]
Also automated buildings of shops that can be built in plots ? (like brewery or hospital ran by villagers)
Trading is not allways imaginary.. sometimes horses with minecarts and soliders go thru one nation to
other one.. so better protect them from barbarians..
Rage/anger boost? (if a nation became a traitor and became allies with your enemy nation you get rage powers when trying to invade that nation)
Slavery/jailing (like being able to jail/slave people to work in jails/mines/dirty jobs) (with a possible chance of escaping ?)
Nation-wise mines? (like prison servers, auto replacing but ran by nations and mostly traitors/captives/prisoners/slaves work here)
a rich/wealthy rank in between member and councilor (can open/run shops and gain some percent money from mines)
a rank for soldiers in nations (may have homes and may get paid by nation)
Plots in other worlds? (nether nations,end nations)
World based technology tree? (so nether nations have a different tech tree than end.)
A counter attack way? (so the nation soldiers can counter attack sieges)
Sieges don't have to be imaginary (let them be linked minecarts driven by horses and mostly moved by soldiers to the war. If left behind they mey get counter attacked/destroyed)
Nation-wise luck (more like Luck of the god.. nation wise diseases or wealth boosts maybe strenght boosts)

**Comment by LaxWasHere** [comment]
Whoa. A plot plugin with a faction like features? Waat. I believe this is a _GREAT_ (please make it soon) idea although I think it should be separate from the plugin. An add-on perhaps?

**Comment by EvilOlaf** [comment]
I agree with @LaxWasHere . These are pretty big changes which should be outsourced in an addon.

**Comment by boy0001** [comment]
This was never going to be in the main plugin, it was just an idea that I thought of last year, then ignored. If there's genuine interest then perhaps I can work on it.

**Comment by EvilOlaf** [comment]
As for me I don't need it. Others may like it though.
